File: tuan2/PhanA/bai36.cpp
// C++ (bai36.cpp)
#include <iostream>
#include <climits>

using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n;
    if (!(cin >> n) || n < 0) return 0;

    long long f = 1;
    bool overflow = false;

    for (int i = 2; i <= n; ++i) {
        // Kiểm tra trước khi nhân để phát hiện tràn số
        if (f > LLONG_MAX / i) {
            overflow = true;
            break;
        }
        f *= i;
    }

    if (overflow) {
        cout << "TRAN SO\n";
    } else {
        cout << f << "\n";
    }

    return 0;
}

File: tuan2/PhanA/bai36.py
# Python (bai36.py)
import sys

# Giới hạn lớn nhất của số nguyên 64-bit có dấu (tương đương LLONG_MAX trong C++)
LLONG_MAX = 9223372036854775807

def main():
    s = sys.stdin.read().strip()
    if not s: return
    n = int(s)
    if n < 0: return

    f = 1
    overflow = False

    for i in range(2, n + 1):
        if f > LLONG_MAX // i:
            overflow = True
            break
        f *= i

    if overflow:
        print("TRAN SO")
    else:
        print(f)

if __name__ == "__main__":
    main()